portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -45,10 +50,29 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
       long id = entity.getKey().getId();
       String text = (String) entity.getProperty("text");
       String author = (String) entity.getProperty("author");
-      Comment comment = new Comment(id, text, author);
+      String mood = (String) entity.getProperty("mood");
+      Long timestamp = (Long) entity.getProperty("timestamp");
+
+      Comment comment = new Comment(id, text, author, mood, timestamp);
       comments.add(comment);
     }
 
+    /* Create filters for sorting the comments */
```

Comment:
You could consider factoring out a method that does the sorting. This shortens the method `doGet` and makes it easier to read.